### PR TITLE
Add ln_sf function for soft link creation

### PR DESCRIPTION
**Pull request by adamgilbert**

### CODE REVIEW

File: shtdlib.sh [hunk 1]
```diff
@@ -1524,6 +1524,30 @@ function load_config()
...
+        else
+            debug 6 "Current symlink at ${link_path} already points to ${target_path}"
+            return 0
+        fi
+    elif [ -e "${link_path}" ]; then
+        color_echo red "Found filesystem object at: ${link_path} but it's not a symlink, fatal error, exiting!"
+        exit_on_fail
+    fi
+    # Create symlink
+    ln -s "${target_path}" "${link_path}"
```
Comment (hrpatel) [inline shtdlib.sh]: Optional... Should we add a debug statement just before returning to say that the symlink was successfully created?

File: shtdlib.sh [hunk 2]
```diff
@@ -1524,6 +1524,30 @@ function load_config()
     done < "${config_file}";
 }
 
+# Make sure symlink exists and points to the correct target, will remove
+# symlinks pointing to other locations or do nothing if it's correct.
+function ln_sf {
+    target_path="${1}"
+    link_path="${2}"
```
Comment (mroote) [inline shtdlib.sh]: Should we return an error if the params aren't provided?

Comment (hrpatel) [reply shtdlib.sh]: Good point.. we can `assert` that both are non-empty

Comment (adamgilbert) [reply shtdlib.sh]: I will update to include a quick args count check.

**Comment by mroote** [comment]
+1

**Comment by hrpatel** [comment]
+1

**Pull request merged**